### PR TITLE
feat(workspace/warehouse_app): add resource to manage warehouse application

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add a new resource used to add application to Workspace app warehouse.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage application.
2. add related document, environmentvariable and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
./scripts/coverage.sh -o workspace -f TestAccAppWarehouseApp_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccAppWarehouseApp_basic -timeout 360m -parallel 10
=== RUN   TestAccAppWarehouseApp_basic
=== PAUSE TestAccAppWarehouseApp_basic
=== CONT  TestAccAppWarehouseApp_basic
--- PASS: TestAccAppWarehouseApp_basic (34.29s)
PASS
coverage: 5.2% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 34.487s coverage: 5.2% of statements in ./huaweicloud/services/workspace
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found 
image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/workspace_app_warehouse_app.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+subcategory: "Workspace"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_workspace_app_warehouse_app"
+description: |-
+  Use this resource to add a application to Workspace APP warehouse within HuaweiCloud.
+---
+
+# huaweicloud_workspace_app_warehouse_app
+
+Use this resource to add a application to Workspace APP warehouse within HuaweiCloud.
```
Comment (Lance52259) [inline docs/resources/workspace_app_warehouse_app.md]:
```suggestion
Manages an application resource of Workspace APP warehouse within HuaweiCloud.
```

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o workspace -f TestAccAppWarehouseApp_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccAppWarehouseApp_basic -timeout 360m -parallel 10
=== RUN   TestAccAppWarehouseApp_basic
=== PAUSE TestAccAppWarehouseApp_basic
=== CONT  TestAccAppWarehouseApp_basic
--- PASS: TestAccAppWarehouseApp_basic (31.51s)
PASS
coverage: 5.2% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 31.582s coverage: 5.2% of statements in ./huaweicloud/services/workspace

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o workspace -f TestAccAppWarehouseApp_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccAppWarehouseApp_basic -timeout 360m -parallel 10
=== RUN TestAccAppWarehouseApp_basic
=== PAUSE TestAccAppWarehouseApp_basic
=== CONT TestAccAppWarehouseApp_basic
--- PASS: TestAccAppWarehouseApp_basic (31.51s)
PASS
coverage: 5.2% of statements in ./huaweicloud/services/workspace
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 31.582s coverage: 5.2% of statements in ./huaweicloud/services/workspace

**Pull request merged**